### ISSUE TITLE
Dropdown is missing critical accessibility attributes

### ISSUE THREAD

**Issue by tomekregulski** [ant-design/ant-design#48132]
### What problem does this feature solve?

Currently the Dropdown component does not meet WAI-ARIA standards for Menu Button Pattern (https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/):

Specifically:
- The element with role button has aria-haspopup set to either menu or true.
- When the menu is displayed, the element with role button has aria-expanded set to true. When the menu is hidden, aria-expanded is set to false.
- The element that contains the menu items displayed by activating the button has role menu.
Optionally, the element with role button has a value specified for aria-controls that refers to the element with role menu.

And furthermore from the Menu and MenuBar Pattern (https://www.w3.org/WAI/ARIA/apg/patterns/menubar/), though neither of these roles are used in Dropdown, it can be reasonably deduced that in Dropdown, the element with role=button should accept an aria-label prop to assist those with screen readers properly identifying it.

The Dropdown component currently does accept an aria-label prop, but it does not seem to actually get applied anywhere in the DOM structure.

### What does the proposed API look like?

Most of the above is satisfied by internal definitions, but the Dropdown component must expose an aria-label prop that maps to the aria-label attribute on the element that has role=button.

**Comment by KirilCycle** [comment]
@afc163 Hello can i try this ? This task looks fascinating

**Comment by afc163** [comment]
@KirilCycle Sure.

**Comment by KirilCycle** [comment]
@afc163 Hello! The focus doesn't land on the first menu item. It requires an additional tab press to activate it. This behavior is not recommended, but it's not a dealbreaker either. I'm finding it a bit tricky to set focus on the first element because the menu ref is null before the menu is mounted. Can you offer any hints, please ?
